h5io/_h5io.py

```diff
@@ -123,19 +150,32 @@ def _triage_write(key, value, root, comp_kw, where):
         _create_titled_dataset(root, key, 'ndarray', value)
     elif sparse is not None and isinstance(value, sparse.csc_matrix):
         sub_root = _create_titled_group(root, key, 'csc_matrix')
-        _triage_write('data', value.data, sub_root, comp_kw,
-                      where + '.csc_matrix_data')
-        _triage_write('indices', value.indices, sub_root, comp_kw,
-                      where + '.csc_matrix_indices')
-        _triage_write('indptr', value.indptr, sub_root, comp_kw,
-                      where + '.csc_matrix_indptr')
+        cleanup_data = _triage_write(
+            'data', value.data, sub_root, comp_kw,
+            where + '.csc_matrix_data', cleanup_data=cleanup_data)
+        cleanup_data = _triage_write(
+            'indices', value.indices, sub_root, comp_kw,
+            where + '.csc_matrix_indices', cleanup_data=cleanup_data)
+        cleanup_data = _triage_write(
+            'indptr', value.indptr, sub_root, comp_kw,
+            where + '.csc_matrix_indptr', cleanup_data=cleanup_data)
     else:
-        raise TypeError('unsupported type %s (in %s)' % (type(value), where))
-
+        if isinstance(value, (DataFrame, Series)):
+            if isinstance(value, DataFrame):
+                title = 'pd_dataframe'
+            else:
+                title = 'pd_series'
+            rootname = root.name
+            cleanup_data.append({title: (rootname, key, value)})
+        else:
+            err_str = 'unsupported type %s (in %s)' % (type(value), where)
+            raise TypeError(err_str)
+    return cleanup_data
```

Comment:
Note that there's nothing functionally wrong with what you did, but it's cleaner not to return anything because it makes it clearer it's modified in place. In other words, the return / reassignment doesn't really do anything. By omitting it, then, you make it clearer the object must be getting modified in place.